platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/TaskContainerExtensions.kt

```diff
@@ -276,5 +276,58 @@ inline fun <reified T : Task> TaskContainer.register(name: String, vararg argume
  * Creates a [Task] with the given [name] and type, passing the given arguments to the [javax.inject.Inject]-annotated constructor,
  * and adds it to this project tasks container.
  */
+@Deprecated("Use register instead. See https://docs.gradle.org/current/userguide/task_configuration_avoidance.html for more information.", ReplaceWith("this.register<T>(name, arguments)"))
+@Suppress("DEPRECATION")
 inline fun <reified T : Task> TaskContainer.create(name: String, vararg arguments: Any): T =
```

Comment:
From my local testing, the `create` overload with varargs is not being replaced correctly
From this code:
```
tasks.create<MyTask>("foo", "hey")
```
Applying the replacement yields:
```
tasks.register<MyTask>("foo", arrayOf<Any>("hey"))
```
Instead of having a single vararg argument of type `String`, now there is a single vararg argument of type `Array<Any>`.

It's probably not a show stopper, but users _will_ get a build error (potentially later, because of laziness), because it's very unlikely that the task constructor will be happy with its first argument being replaced by an array.

Reply:
I found a fix: we need to use the spread-operator in the replace-with expression:
```suggestion
@Deprecated("Use register instead. See https://docs.gradle.org/current/userguide/task_configuration_avoidance.html for more information.", ReplaceWith("this.register<T>(name, *arguments)"))
@Suppress("DEPRECATION")
inline fun <reified T : Task> TaskContainer.create(name: String, vararg arguments: Any): T =
```

Reply:
Fixed in two places